### PR TITLE
#184347041 retrieve order status

### PR DESCRIPTION
**Pull request by Aline-Uwera**
### What does this PR do?
Buyer should be able to track their order status
### Description of Task to be completed?
- Buyers can be able to view their order information and order status
- Buyers can get sales associated to their order and get each sale status
- Sellers can be able to get all their sales

### How should this be manually tested?
1. Checkout to branch `feature/track-order-status-184347041`
2. Open browser and visit `http://localhost:<5000>/api-docs/`
3. Test functionality of order and sales endpoints

#### What are the relevant pivotal tracker stories?
[184347041](https://www.pivotaltracker.com/story/show/184347041)

### CODE REVIEW

**Review by irakozetony — APPROVED** [review]
Change the commit message

File: src/controllers/order.controller.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { OrderService } from '../services/order.service';
+
+export class OrderController {
+    static async getOrder(req, res) {
+        try {
+            const buyerId = req.user.id;
+            const order = await OrderService.getBuyerOrder(buyerId);
+            return res.status(200).json({ order });
```
Comment (nshimiyejayd) [inline src/controllers/order.controller.js]: Is this the way to return JSON?

File: src/controllers/sales.controller.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        try {
+            const orderId = req.order.id;
+            const sales = await SalesService.getOrderSales(orderId);
+            return res.status(200).json({ message: 'All order sales retrieved successfully', payload: sales.productId });
+        } catch (error) {
+            return res.status(500).json({
+                error: error.message,
+                message: 'Could not retrieve sales, try again',
+            });
+        }
```
Comment (fabmukunzi) [inline src/controllers/sales.controller.js]: How a seller will retrieve all sales that belongs to him/her

File: src/routes/api/sales.routes.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+const salesRoutes = express.Router();
+
+salesRoutes
+    .route('/:saleId/saleStatus')
+    .get(
+        protectRoute,
+        restrictTo('seller', 'buyer', 'admin'),
+        checkIfSaleExists,
+        SalesController.getSaleStatus
+    );
```
Comment (fabmukunzi) [inline src/routes/api/sales.routes.js]: will a seller be able to retrieve status of other seller's sales

**Review by nshimiyejayd — APPROVED** [review]

**Review by fabmukunzi — APPROVED** [review]

**Comment by ncutixavier** [comment]
- @Aline096 Please update the PR description to mention all tasks you have done
- Please remove the dash from your commit message

**Review by ncutixavier — APPROVED** [review]

**Review by istafrica-dennw — APPROVED** [review]

**Review by ncutixavier — COMMENTED** [review]
Resolve the conflicts

**Review by ncutixavier — APPROVED** [review]

**Pull request merged**